### PR TITLE
[MFTF] Refactoring StorefrontVerifyOrderHistoryCommentsTest

### PR DESCRIPTION
**Pull request by AnnaAPak**
The test is refactored to improve execution time

### Related Pull Requests
https://github.com/magento/magento2/pull/33557

**Please, note:** the test is dependent on https://github.com/magento/magento2/pull/33557

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCancelTheCreatedOrderWithCashOnDeliveryPaymentMethodTest.xml [hunk 1]
```diff
@@ -43,39 +49,34 @@
...
-        <click selector="{{StorefrontCustomerSidebarSection.sidebarCurrentTab('My Orders')}}" stepKey="clickOnMyOrders"/>
-        <waitForPageLoad stepKey="waitForOrderDetailsToLoad"/>
-        <seeElement selector="{{StorefrontCustomerOrderViewSection.orderStatusInGrid('$orderId', 'Canceled')}}" stepKey="seeOrderStatusInGrid"/>
+    
+        <getData entity="GetOrderData" stepKey="getOrderData">
+            <requiredEntity createDataKey="createCustomerCart"/>
+        </getData>
+        <actionGroup ref="StorefrontOpenMyOrdersPageActionGroup" stepKey="clickOnMyOrders"/>
+        <comment userInput="Comment is added to preserve the step key for backward compatibility" stepKey="waitForOrderDetailsToLoad"/>
+        <seeElement selector="{{StorefrontCustomerOrderViewSection.orderStatusInGrid('$createCustomerCart.increment_id$', 'Canceled')}}" stepKey="seeOrderStatusInGrid"/>
```
Comment (edenduong) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCancelTheCreatedOrderWithCashOnDeliveryPaymentMethodTest.xml]: I think can move to action group with argument

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontOrderCommentWithHTMLTagsDisplayTest.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            <argument name="comment" value="{{SalesOrderComments.commentWithHyperlink}}" />
+        </actionGroup>
+
+        <actionGroup ref="LoginToStorefrontActionGroup" stepKey="loginToStorefront">
+            <argument name="Customer" value="$$createCustomer$$"/>
+        </actionGroup>
+
+        <actionGroup ref="StorefrontClickViewOrderLinkOnMyOrdersPageActionGroup" stepKey="clickViewOrder"/>
+
+        <dontSee userInput="{{anchorTagFragment.anchorTag}}" selector="{{StorefrontCustomerOrderViewSection.orderComment}}" stepKey="dontSeeExposedHtmlCode"/>
```
Comment (edenduong) [inline app/code/Magento/Sales/Test/Mftf/Test/StorefrontOrderCommentWithHTMLTagsDisplayTest.xml:62]: I think can move to action group

**Review by edenduong — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please check my comment

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCancelTheCreatedOrderWithCashOnDeliveryPaymentMethodTest.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Sales/Test/Mftf/Test/AdminCancelTheCreatedOrderWithCashOnDeliveryPaymentMethodTest.xml]: Hi @edenduong I have created the separate PR for this Action Group - please check https://github.com/magento/magento2/pull/33727.
I have updated other tests with it as well.

These changed are within the initial PR and there are some more PR depended on it.

Please let me know if you are ok with such approach.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontOrderCommentWithHTMLTagsDisplayTest.xml (hunk 2 above)
Comment (AnnaAPak) [reply app/code/Magento/Sales/Test/Mftf/Test/StorefrontOrderCommentWithHTMLTagsDisplayTest.xml:62]: @edenduong I checked - there are no more tests that use this assertion. Do you think we still need create a separate Action Group here?
Thanks.

Comment (Den4ik) [reply app/code/Magento/Sales/Test/Mftf/Test/StorefrontOrderCommentWithHTMLTagsDisplayTest.xml:62]: @AnnaAPak It could be groundwork for the future :)

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento run Functional Tests CE,Functional Tests B2B

**Review by sidolov — APPROVED** [review]

**Review by Den4ik — APPROVED** [review]

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by engcom-Charlie** [comment]
Hi @AnnaAPak,

Thank you for your contribution!

Will look into the conflicts.

Thank you!

**Comment by engcom-Charlie** [comment]
It seems that Bot moved this ticket to Ready for testing automatically so moving it back to Change Requested.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Charlie** [comment]
I have merged 2.4-develop into this branch and ran the tests. 

Since the build is green now and PR is about MFTF test cases, moving this directly to Merge in Progress.

Thank you!

**Pull request merged**